hypershift-operator/metrics.go

```diff
@@ -485,6 +504,15 @@ func (m *hypershiftMetrics) observeNodePools(ctx context.Context, nodePools *hyp
 	return nil
 }
 
+func (m *hypershiftMetrics) observeOperator(ctx context.Context, hypPod corev1.Pod) error {
+	hypVersion := version.GetRevision()
+	hypImage := hypPod.Spec.Containers[0].Image
+	hypImageID := hypPod.Status.ContainerStatuses[0].ImageID
```

Comment:
I don't think we can guarantee [0], can we instead search by the container name i.e. operator?